apps/server/src/app.ts

```diff
@@ -324,3 +330,46 @@ process.on('uncaughtException', async (error) => {
 process.once('SIGHUP', async () => shutdown(0));
 process.once('SIGINT', async () => shutdown(0));
 process.once('SIGTERM', async () => shutdown(0));
+
+/**
+ * @description tries to open the server with the desired port, and if getting a `EADDRINUSE` will change to an efemeral port
+ * @param {http.Server}server http server object
+ * @param {number}desiredPort the desired port
+ * @returns {number} the resulting port number
+ * @throws any other server errors will result in a throw
+ */
+async function serverTryDesiredPort(server: http.Server, desiredPort: number): Promise<number> {
+  return new Promise((res) => {
+    expressServer.once('error', (e) => {
+      if (isDocker) throw e; // we should only move ports if we are in a desktp environment
```

Comment:
```suggestion
      if (isDocker) throw e; // we should only move ports if we are in a desktop environment
```

Also, question: will Ontime Cloud be have the NODE_ENV set to `docker`? I could actually not find where do we set this 😅 . I wonder if we need a more reliable way to check if we are in electron

Reply:
it is set in the docker build file
yes, a isElectron would be good, where could we set that?